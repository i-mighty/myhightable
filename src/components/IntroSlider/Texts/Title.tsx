import React from 'react';
import styled from 'styled-components/native';

export interface TitleProps {
  color?: string;
  text: string;
  testID?: string;
}

const StyledTitle = styled.Text<Pick<TitleProps, 'color'>>`
  font-family: Roboto;
  font-style: normal;
  font-weight: bold;
  font-size: 24px;
  line-height: 28px;
  text-align: center;
  color: ${({ color, theme }) => (color ? color : theme.vars.orange)};
  margin-bottom: 5px;
`;

const Title: React.FC<TitleProps> = ({ text, testID, color }) => {
  return (
    <StyledTitle color={color} testID={testID}>
      {text}
    </StyledTitle>
  );
};

export default Title;
