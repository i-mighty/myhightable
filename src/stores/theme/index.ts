interface Theme {
  vars: {
    [key: string]: string;
  };
  dimens?: {
    [key: string]: string;
  };
}

const theme: Theme = {
  vars: {
    orange: '#F89505',
    dark: '#000000',
    black: '#333333',
    black_alt: '#4F4F4F',
    slate_grey: '#828282',
    grey: '#BDBDBD',
    grey_alt: '#C4C4C4',
    white: '#FFFFFF',
    white_alt: '#F2F2F2',
  },
};

export default theme;
