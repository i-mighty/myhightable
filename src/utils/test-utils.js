import { ThemeProvider } from 'styled-components/native';
import React from 'react';
import { render as baseRender } from '@testing-library/react-native';
import theme from '@src/stores/theme';

export const renderWithProps = (Component, props) => {
  baseRender(
    <ThemeProvider theme={theme}>
      <Component {...props} />
    </ThemeProvider>,
  );
};
