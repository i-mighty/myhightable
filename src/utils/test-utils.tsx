import { ThemeProvider } from 'styled-components/native';
import React, { ReactElement } from 'react';
import { render as baseRender } from '@testing-library/react-native';
import theme from '@src/stores/theme';

export const renderWithProps = (
  Component: React.ElementType,
  props: Object,
) => {
  return baseRender(
    <ThemeProvider theme={theme}>
      <Component {...props} />
    </ThemeProvider>,
  );
};
