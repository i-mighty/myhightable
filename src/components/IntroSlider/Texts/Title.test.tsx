import Title from './Title';
import theme from '@src/stores/theme';
import { renderWithProps } from '@src/utils/test-utils';

const defaultProps = { testID: 'title', text: 'HighTable' };
const colouredProps = { ...defaultProps, color: theme.vars.white };

describe('Title', () => {
  it('renders successfully', () => {
    const { getByTestId, baseElement } = renderWithProps(Title, defaultProps);
    const textComponent = getByTestId(defaultProps.testID);
    expect(textComponent).toHaveStyleRule('color', theme.vars.orange);
    expect(getByTestId(defaultProps.testID).props.children).toEqual(
      defaultProps.text,
    );
    expect(baseElement).toMatchSnapshot();
  });
  it('renders with the right color', () => {
    const { getByTestId, baseElement } = renderWithProps(Title, colouredProps);
    const textComponent = getByTestId(colouredProps.testID);
    expect(textComponent.props.children).toEqual(colouredProps.text);
    expect(textComponent).toHaveStyleRule('color', colouredProps.color);
    expect(baseElement).toMatchSnapshot();
  });
});
