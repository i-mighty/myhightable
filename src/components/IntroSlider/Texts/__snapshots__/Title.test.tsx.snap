// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title renders successfully 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#F89505",
            "fontFamily": "Roboto",
            "fontSize": 24,
            "fontStyle": "normal",
            "fontWeight": "bold",
            "lineHeight": 28,
            "marginBottom": 5,
            "textAlign": "center",
          },
        ]
      }
      testID="title"
    >
      HighTable
    </Text>
  </View>
</View>
`;

exports[`Title renders with the right color 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      color="#FFFFFF"
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontFamily": "Roboto",
            "fontSize": 24,
            "fontStyle": "normal",
            "fontWeight": "bold",
            "lineHeight": 28,
            "marginBottom": 5,
            "textAlign": "center",
          },
        ]
      }
      testID="title"
    >
      HighTable
    </Text>
  </View>
</View>
`;
